src/lib/serialization/components.ts

```diff
@@ -1,20 +1,14 @@
-import { Reflection, Type } from "../models";
-
 import type { Serializer } from "./serializer";
 import type { ModelToObject } from "./schema";
 
 /**
  * Represents Serializer plugin component.
  *
  * Like {@link Converter} plugins each {@link Serializer} plugin defines a predicate that instructs if an
- * object can be serialized by it, this is done dynamically at runtime via a `supports` method.
+ * object can be serialized by it. This is done dynamically at runtime via a `supports` method.
  *
  * Additionally, each {@link Serializer} plugin must define a predicate that instructs the group
  * it belongs to.
- *
- * Serializers are grouped to improve performance when finding serializers that apply to a node,
- * this makes it possible to skip the `supports` calls for `Type`s when searching for a
- * `Reflection` and vise versa.
  */
 export abstract class SerializerComponent<T> {
```

Comment:
I think this can just become a non-generic interface now - never liked the horrible static `PRIORITY` thing and it only lived this long because of backwards compatibility.

Reply:
`toObject` still takes the generic's type, so I left it for now. I did however convert `SerializerComponent` to an interface.